config/crd/bases/test.openstack.org_tempests.yaml

```diff
@@ -168,6 +168,10 @@ spec:
                     description: A content of exclude.txt file that is passed to tempest
                       via --exclude-list
                     type: string
+                  expectedFailuresList:
+                    description: A content of expected_failures.txt file that is passed
+                      to tempest via --expected-failures-list
```

Comment:
@kopecmartin do you think `--expected-failures-list` is something that would be "mergeable" in tempest.

Reply:
this doesn't belong to upstream tempest, tempest tests are meant to be explicit - https://opendev.org/openstack/tempest/src/branch/master/REVIEWING.rst#being-explicit
Parameter like this may mask failures (especially if a user forgets that a test result is being ignored) because they'll be ignored .. if tests fail due to a known bug, the user should not run them - if it's a bug in tempest, it can be skipped in tempest, if it's a bug in a component, well, the test did its job - if a user doesn't want a failing CI in the meantime, they can exclude such test.

that's an upstream perspective ^^

Downstream perspective:
I'm still not convinced this is really needed as any failing or potentially failing test can be simply excluded.

Reply:
This have nothing to do with tempest in upstream. I expect to have a list of tests that I know it's unstable (timeout, concurrency issues, etc) and I want testoperator to check if the match list of expected failing tests matches the failing tests in a particular job, and if does, don't mark the job as failure. The tempest test will continue to fail, and that is okay, but in CI, the job will be marked as success and we have the failures reported to report portal/polarion. This is important to add unijobs into the promotion criteria, since unijobs run 5k tests, sometimes a test fails because of concurrency, or whitebox tests for example that are very unstable. But we want to collect the data of how many times it fails/pass so we can decide later what to do with these tests.

Reply:
all clear now, this in test-operator is gonna be an interface how to get the tests (that might fail) to tcib script where the "don't fail on these tests" logic will be written:
https://github.com/openstack-k8s-operators/tcib/blob/main/container-images/tcib/base/os/tempest/run_tempest.sh

this won't reach the tempest upstream.

Reply:
Exactly, this is a unijobs downstream feature only. Nothing related to upstream tempest.

Reply:
Ok, it seems we have an agreement and common understanding of how it should work. In this case just Katka, please, when you have time and this PR is ready, update the description of the `expectedFailuresList`. Right now it suggests that the `--expected-failures-list` parameter is passed to tempest.

Reply:
Thank you @arxcruz for making the intention for this parameter clear. It is my mistake for writing the description in such a way because I followed the code in include and exclude list. I fixed the description in the last push, so I hope from now on it will be smooth sailing. Sorry for the confusion :sweat_smile:

Reply:
That's ok, @kstrenkova. At least we sorted things out :)